### ISSUE TITLE
Move pretty print button into editor, bottom right

### ISSUE THREAD

**Issue by helenvholmes** [jlongster/debugger.html#229]
Needs a white background and editor needs a few extra lines appended to the bottom so it doesn't obscure the code at the bottom.

**Comment by jasonLaster** [comment]
Mentioning this because it wasnt obvious to me. The pretty print button is currently in the top right of the sidebar.

We should add a footer element below the codemirror editor, that can hold the pretty print button. The footer will be similar to how source tabs are shown as well.

**Comment by helenvholmes** [comment]
I had actually imagined this a little differently:

Instead of a footer, I was imagining the pretty print button to hover over the editor in the bottom right hand corner. This will definitely obscure some code at any one time, but not a lot. In order to ensure that people can always see all of the code, I want us to add a few extra lines to the editor so that one can scroll past ones' code and see the editor below the pretty print button.

@jasonLaster I think this will ultimately make for a cleaner UI, but is this problematic in any way I haven't thought of?

**Comment by jlongster** [comment]
I like the idea of not having a whole footer (for now at least, since there's nothing else we'll put in there), and we should diverge from Chrome where we can. I don't care about copying general good UI patterns, but it's going to be easy to just look like we're copying them completely.

**Comment by helenvholmes** [comment]
For reference, looks like Chrome uses their footer to give the user info about their cursor position:

![screen shot 2016-06-03 at 9 47 20 am](https://cloud.githubusercontent.com/assets/1720093/15780533/2ebaef1e-2970-11e6-9903-31591a7bdf0c.png)

I don't think our current designs have this anywhere—might be worth thinking about how/if we should include this info.

**Comment by jasonLaster** [comment]
Ahh, I didn't realize that. 

I like the idea of pretty print floating in the bottom right. 

I can see us using the right side of the editor down the road, but I don't think that should stop us now. 

We might also want to reconsider if we want more current source actions like pretty print, but I think we're okay. 

I am not too worried about cursor info in most cases, the one case I can think of is minified code where column info is important.

**Comment by jasonLaster** [comment]
One thought, it would be nice to support "go to line". Maybe we could reuse the file search top bar.

**Comment by jasonLaster** [comment]
I started this work here: https://github.com/jlongster/debugger.html/compare/master...jasonLaster:editor-btns?expand=1

Happy to let someone pick it up and see it through.

I noticed while playing with it that overlaying the buttons as a bottom bar looked awkward while scrolling through a file. It was not clear how much opacity to add or how wide the bar should be.

It might be easier to move the footer below the editor so that it is a little more readable.
